src/server/list_family.cc

```diff
@@ -1173,8 +1174,165 @@ void BPopGeneric(ListDir dir, CmdArgList args, Transaction* tx, SinkReplyBuilder
   return rb->SendNullArray();
 }
 
+struct LMPopParams {
+  uint32_t num_keys;
+  ListDir dir;
+  int pop_count;
+};
+
+// Validates the LMPop command arguments and extracts the values to the output params.
+// If the arguments are invalid sends the appropriate error to builder and returns false.
+bool ValidateLMPopCommand(CmdArgList args, LMPopParams* params, SinkReplyBuilder* builder) {
```

Comment:
nit: I think `ParseLMPop` is a better name (parse also implies validating).

Also, we have a type  `facade::ErrorReply` that can be returned instead of bool. 
it may hold all types of errors you return here (we had an evolution of types for historical reasons).
This way you do not need to pass `SinkReplyBuilder* builder` to this function and instead check if `ErrorReply::status == kOk` in the caller to see if parsing succeeded.